Improve app signature check

Isira-Seneviratne opened the pull request:
<!-- Hey there. Thank you so much for improving NewPipe, and filling out the details. Having roughly the same layout helps everyone considerably :)-->

#### What is it?
- [x] Codebase improvement (dev facing)

#### Description of the changes in your PR
<!-- While bullet points are the norm in this section, feel free to write free-form text instead of a list -->
- Simplify the app signature check using [`PackageInfoCompat.hasSignatures`](https://developer.android.com/reference/androidx/core/content/pm/PackageInfoCompat#hasSignatures(android.content.pm.PackageManager,java.lang.String,java.util.Map%3Cbyte[],java.lang.Integer%3E,boolean)).

#### APK testing
<!-- Use a new, meaningfully named branch. The name is used as a suffix for the app ID to allow installing and testing multiple versions of NewPipe, e.g. "commentfix", if your PR implements a bugfix for comments. (No names like "patch-0" and "feature-1".)  -->
<!-- Remove the following line if you directly link the APK created by the CI pipeline. Directly linking is preferred if you need to let users test.-->
The APK can be found by going to the "Checks" tab below the title. On the left pane, click on "CI", scroll down to "artifacts" and click "app" to download the zip file which contains the debug APK of this PR. You can find more info and a video demonstration [on this wiki page](https://github.com/TeamNewPipe/NewPipe/wiki/Download-APK-for-PR).

#### Due diligence
- [x] I read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md).

AudricV commented on app/src/main/java/org/schabi/newpipe/util/ReleaseVersionUtil.kt:
```diff
@@ -1,97 +1,39 @@
-    private fun showRequestError(app: App, e: Exception, request: String) {
-        createNotification(
-            app, ErrorInfo(e, UserAction.CHECK_FOR_NEW_APP_VERSION, request)
+        @Suppress("NewApi")
+        val certificates = mapOf(
+            RELEASE_CERT_PUBLIC_KEY_SHA256.toByteArray() to PackageManager.CERT_INPUT_SHA256
```
Why are you using a constant available only on Android API 28 and higher and suppressing its availability warning? Is it a thing which would work on all devices thanks to compilation steps and `compileSdk` set currently to Android API 34?

Also, maybe you should move the certificates map as a constant in order to not run the map creation each time the method is called, as it can be called multiple times (see the usages of the `isReleaseApk` method). Let me know what you think about this.

Isira-Seneviratne replied:
> Why are you using a constant available only on Android API 28 and higher and suppressing its availability warning? Is it a thing which would work on all devices thanks to compilation steps and `compileSdk` set currently to Android API 34?

It gets inlined during compilation, as it's an integer value, so it won't cause issues on older devices.

> 
> Also, maybe you should move the certificates map as a constant in order to not run the map creation each time the method is called, as it can be called multiple times (see the usages of the `isReleaseApk` method). Let me know what you think about this.

Yeah, that sounds good.

Isira-Seneviratne commented:
> Did you test your changes by using your own singing key? I tried to do so and it did not work for me. What I did:
> 
> * Replace `RELEASE_CERT_PUBLIC_KEY_SHA256` with the fingerprint of my key
> * build release APK and sign it
> * install signed APK
> * no "Updates" option in the settings, thus `ReleaseVersionUtil.INSTANCE.isReleaseApk()` yields `false` in SettingsActivity.

My bad, I added the certificate SHA-256 digest earlier. I've updated the PR with the public key hash, could you recheck?
